webapp/components/create_post.jsx

```diff
@@ -179,8 +179,10 @@ class CreatePost extends React.Component {
         this.setState({messageText: '', submitting: false, postError: null, previews: [], serverError: null});
 
         Client.createPost(post,
-            () => {
-                // DO nothing. Websockets will handle this.
+            (data) => {
+                if (data) {
+                    PostStore.removePendingPost(data.pending_post_id);
```

Comment:
I think if it was happening then it was caused by something that happens before the websocket event is sent and after the return of the HTTP request but I updated just to be sure